torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,305 @@
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        exists = []
+        for year in self.years:
+            filename_year = self.filename_glob.replace("*", str(year))
+            dirname_year = filename_year.split(".")[0]
+            pathname = os.path.join(self.root, dirname_year, filename_year)
+            if os.path.exists(pathname):
+                exists.append(True)
+            else:
+                exists.append(False)
+
+        if all(exists):
+            return
+
+        # Check if the zip files have already been downloaded
+        exists = []
+        for year in self.years:
+            pathname = os.path.join(
+                self.root, self.zipfile_glob.replace("*", str(year))
+            )
+            if os.path.exists(pathname):
+                exists.append(True)
+                self._extract()
+            else:
+                exists.append(False)
+
+        if all(exists):
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        for year in self.years:
+            download_url(
+                self.url.format(year),
+                self.root,
+                md5=self.md5s[year] if self.checksum else None,
+            )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        for year in self.years:
+            zipfile_name = self.zipfile_glob.replace("*", str(year))
+            pathname = os.path.join(self.root, zipfile_name)
+            extract_archive(
+                pathname, os.path.join(self.root, zipfile_name.split(".")[0])
+            )
+
+    def plot(
+        self,
+        sample: Dict[str, Any],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`RasterDataset.__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        mask = sample["mask"].squeeze().numpy()
+        ncols = 1
+
+        cmap: "np.typing.NDArray[np.int_]" = np.array(
+            [self.cmap[i] for i in range(len(self.cmap))]
+        )
```

Comment:
What I mean is can be use something like [ListedColormap](https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.ListedColormap.html) and use `imshow(..., cmap=cmap)`